ephios/core/templatetags/event_extras.py

```diff
@@ -55,3 +56,18 @@ def confirmed_shifts(user):
         .filter(end_time__gte=datetime.now())
         .order_by("start_time")
     )
+
+
+@register.filter(name="event_signup_state_counts")
+def event_signup_state_counts(event, user):
+    """
+    Return a counter counting states for a users participations for shifts of an event.
+    Uses None as key for no participation info.
+    """
+    counter = collections.Counter()
+    for shift in event.shifts.all():
```

Comment:
is there any chance of caching smth here? this feels like a very big loop over everything